diff --git a/include/android_stub/android/hardware_buffer.h b/include/android_stub/android/hardware_buffer.h
index aa03b4b096e..7a51ede4736 100644
--- a/include/android_stub/android/hardware_buffer.h
+++ b/include/android_stub/android/hardware_buffer.h
@@ -51,6 +51,10 @@

 #include <android/rect.h>

+#if !defined(__BIONIC__) && !defined(__INTRODUCED_IN)
+#define __INTRODUCED_IN(x)
+#endif
+
 __BEGIN_DECLS

 /**
diff --git a/src/gfxstream/guest/vulkan_enc/ResourceTracker.cpp b/src/gfxstream/guest/vulkan_enc/ResourceTracker.cpp
index 4e0dc43c892..b3f8710d244 100644
--- a/src/gfxstream/guest/vulkan_enc/ResourceTracker.cpp
+++ b/src/gfxstream/guest/vulkan_enc/ResourceTracker.cpp
@@ -148,13 +148,13 @@ struct StagingInfo {
     }

     ~StagingInfo() {
-        for (auto stream : streams) {
-            delete stream;
-        }
-
         for (auto encoder : encoders) {
             delete encoder;
         }
+
+        for (auto stream : streams) {
+            delete stream;
+        }
     }

     void pushStaging(CommandBufferStagingStream* stream, VkEncoder* encoder) {